edweiss-app/app/(app)/deck/[id]/card/creation/index.tsx

```diff
@@ -0,0 +1,95 @@
+import RouteHeader from '@/components/core/header/RouteHeader';
+import { ApplicationRoute } from '@/constants/Component';
+
+import TScrollView from '@/components/core/containers/TScrollView';
+import TView from '@/components/core/containers/TView';
+import FancyButton from '@/components/input/FancyButton';
+import FancyTextInput from '@/components/input/FancyTextInput';
+import { callFunction, Collections } from '@/config/firebase';
+import { useDoc } from '@/hooks/firebase/firestore';
+import Memento from '@/model/memento';
+import { router, useLocalSearchParams } from 'expo-router';
+import React, { useState } from 'react';
+
+const CreateCardScreen: ApplicationRoute = () => {
+    const { deckId } = useLocalSearchParams();
+    const [question, setQuestion] = useState("");
+    const [answer, setAnswer] = useState("");
+    const [existedQuestion, setExistedQuestion] = useState(false);
+
+    const deck = useDoc(Collections.deck, deckId as string);
+
+    async function createCard() {
+
+        if (question.length == 0 || answer.length == 0 || existedQuestion)
+            return;
+
+        const isDuplicate = deck?.data.cards.some(card => card.question === question);
```

Comment:
I just had a small remark on this part of the code: in case that the useDoc hook may not be handling a potentially null or undefined value. For example, on Firebase error the code base assume that it's deck is valid.

Reply:
I will definitely check that.